src/core/comm/coll.cc

```diff
@@ -290,34 +277,46 @@ int collFinalize(void)
   assert(coll_inited == true);
   coll_inited = false;
 #ifdef LEGATE_USE_GASNET
-  for (int i = 0; i < MAX_NB_COMMS; i++) { CHECK_MPI(MPI_Comm_free(&mpi_comms[i])); }
+  for (int i = 0; i < mpi_comms.size(); i++) { CHECK_MPI(MPI_Comm_free(&mpi_comms[i])); }
```

Comment:
let's use a range loop here, instead of a C style loop. (i.e., `for (MPI_Comm& mpi_comm : mpi_comms) { ... }`)

Reply:
fixed, will push it after the sapling is back to normal.